packages/amplify-category-auth/index.js

```diff
@@ -188,12 +188,25 @@ async function checkRequirements(requirements, context) {
 async function initEnv(context) {
   const { amplify } = context;
   const { resourcesToBeCreated, resourcesToBeDeleted, resourcesToBeUpdated } = await amplify.getResourceStatus('auth');
+  let toBeCreated = [];
+  let toBeDeleted = [];
+  let toBeUpdated = [];
 
-  resourcesToBeDeleted.forEach((authResource) => {
+  if (resourcesToBeCreated && resourcesToBeCreated.length > 0) {
```

Comment:
Why do we need to do this additonal filtering? The amplify.getResourceStatus('auth') function should already filter your resources based on auth.

Reply:
That's what I would exect as well, but for some reason I'm getting this:
```
resourcesToBeCreated
Array(2) [Object, Object]
length:2
__proto__:Array(0) [, …]
0:Object {service: "Cognito", providerPlugin: "awscloudformation", dependsOn: Array(1), …}
category:"auth"
dependsOn:Array(1) [Object]
providerPlugin:"awscloudformation"
resourceName:"clilaminit8f1d23e0"
service:"Cognito"
__proto__:Object {constructor: , __defineGetter__: , __defineSetter__: , …}
1:Object {service: "Lambda", providerPlugin: "awscloudformation", build: true, …}
build:true
category:"function"
providerPlugin:"awscloudformation"
resourceName:"clilaminit8f1d23e0PostConfirmation"
service:"Lambda"
__proto__:Object {constructor: , __defineGetter__: , __defineSetter__: , …}
```

I wasn't entirely sure what the impacts of changing getResourceStatus would be, so I thought it safer to do filtering on the auth side.

Reply:
It already does the filtering for you..checkout the function and the params it expects.